### ISSUE TITLE
Review - Use Case for isPresent() function

### ISSUE THREAD

**Issue by RossMurr4y** [hamlet-io/engine#1461]
## Expected Behaviour

Attribute can receive configuration from a `DeploymentProfile` whilst still respecting if it's enabled or not. The configuration should only enable a given feature if it explicitly defines it as such.

## Current Behaviour

Currently, the use of the `isPresent()` function within the engine is used to establish if a solution attribute is considered _both_ enabled _and_ configured. A number of state-routines create resources based on this logic, and then further assumptions are made in the setup-routine based on those resources existing. However the `<attribute>.Enabled` will apply by default to any attribute that has configuration. The result of this of applying partial configuration via `DeploymentProfile` will also enable a feature, possibly unintentionally.

## Possible Solution

- define `<attribute>.Enabled` as a metaparameter so it is documented in all the places it can exist
- revisit uses of `isPresent` and consider checking the `Enabled` metaparameter instead.

**Comment by roleyfoley** [comment]
>     * define `<attribute>.Enabled` as a metaparameter so it is documented in all the places it can exist 
>     * revisit uses of `isPresent` and consider checking the `Enabled` metaparameter instead.

It might be better to make Enabled an explicit parameter where its required, not all subobjects support the use of Enabled and this would ensure that we have implemented it where we need to

**Comment by RossMurr4y** [comment]
> > ```
> > * define `<attribute>.Enabled` as a metaparameter so it is documented in all the places it can exist 
> > * revisit uses of `isPresent` and consider checking the `Enabled` metaparameter instead.
> > ```
> 
> It might be better to make Enabled an explicit parameter where its required, not all subobjects support the use of Enabled and this would ensure that we have implemented it where we need to

certainly, I'm still unclear as to where it is and isn't available hence the thought it was "everywhere". That'll be very helpful to have listed.
